### ISSUE TITLE
parallel file.lines()

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#4959]
We would like to have a parallel iterator that traverses a file, one line at a time.

Some initial code towards this goal is here:

https://github.com/tzakian/chapel/blob/bec7f05392141a508ed8042952c3e20c1375facf/modules/standard/ParSplit.chpl

and it is related to the HDFSIterator module.

**Comment by bradcray** [comment]
I had a similar feature request when working on the c-ray ray tracer and created a little wrapper around readline() called readlines() and stuck it in the file itself.  It's only serial, though (and that's all I really wanted in that case).  I've been wanting/meaning to promote this to the standard IO library since writing it.  Thoughts on it?

**Comment by mppf** [comment]
Right, you added (effectively) a channel.lines() but I think file.lines() is necessary to work in parallel (or to work distributed). I think channel.lines() is fine but that we should expect it to always be serial.

**Comment by jabraham17** [comment]
closing, https://github.com/chapel-lang/chapel/pull/25770 added a parallel fileReader.lines
